dev-packages/cli/src/commands/check-header.ts

```diff
@@ -0,0 +1,315 @@
+    const { currentStartYear, expectedStartYear, currentEndYear } = result;
+    result.violation = 'invalidCopyrightYear';
+    result.severity = 'error';
+
+    if (!currentEndYear) {
+        if (currentStartYear === expectedStartYear) {
+            result.violation = 'none';
+            result.severity = 'ok';
+        }
+        return;
+    }
+
+    // Cornercase: For files of the initial contribution the copyright header  predates the first git modification date.
+    // => declare as warning if not part of the initial contribution.
+    if (expectedStartYear === currentEndYear && currentStartYear < expectedStartYear) {
+        if (getFirstCommit(result.file) === getInitialCommit()) {
+            result.violation = 'none';
+            result.severity = 'ok';
+        } else {
+            result.severity = 'warn';
+        }
+    }
+}
+
+function validateTimePeriod(result: DateValidationResult): void {
+    const { currentStartYear, expectedStartYear, expectedEndYear, currentEndYear } = result;
+
+    result.violation = 'incorrectCopyrightPeriod';
+    result.severity = 'error';
+    if (!currentEndYear) {
+        result.severity = 'error';
+        return;
+    }
+
+    if (currentStartYear === expectedStartYear && currentEndYear === expectedEndYear) {
+        result.violation = 'none';
+        result.severity = 'ok';
+        return;
+    }
+
+    // Cornercase: For files of the initial contribution the copyright header predates the first git modification date.
+    // => declare as warning if not part of the initial contribution.
+    if (currentEndYear === expectedEndYear && currentStartYear < expectedEndYear) {
+        if (getFirstCommit(result.file) === getInitialCommit()) {
+            result.violation = 'none';
+            result.severity = 'ok';
+        } else {
+            result.severity = 'warn';
+        }
+    }
+}
+
+function printValidationProgress(currentFileCount: number, maxFileCount: number, file: string): void {
+    process.stdout.clearLine(0);
+    process.stdout.cursorTo(0);
+    process.stdout.write(`[${currentFileCount} of ${maxFileCount}] Validating ${file}`);
+}
+
+export function displayValidationResult(rootDir: string, results: ValidationResult[], options: HeaderCheckOptions): void {
+    LOGGER.newLine();
+    LOGGER.info(`Header validation for ${results.length} files completed`);
+    const violations = results.filter(result => result.severity === 'error');
+    LOGGER.info(`Found ${violations.length} copyright header violations:`);
+    LOGGER.newLine();
+
+    // Adjust results to print based on configured severity level
+    let toPrint = results;
+    if (options.severity === 'error') {
+        toPrint = violations;
+    } else if (options.severity === 'warn') {
+        toPrint = results.filter(result => result.severity !== 'ok');
+    }
+    toPrint.forEach((result, i) => LOGGER.info(`${i}. `, result.file, ':', toPrintMessage(result)));
```

Comment:
Not really an issue but the output will show the first line as `0.` so maybe we should print `i+1` here.